include/deal.II/fe/fe_nedelec_sz.h

```diff
@@ -0,0 +1,426 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2015 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii__fe_nedelec_sz_h
+#define dealii__fe_nedelec_sz_h
+
+#include <deal.II/base/derivative_form.h>
+#include <deal.II/base/qprojector.h>
+#include <deal.II/base/polynomials_integrated_legendre_sz.h>
+#include <deal.II/fe/fe.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/fe/mapping.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/*!@addtogroup fe */
+/*@{*/
+
+/**
+* This class represents an implementation of the
+* H<sup>curl</sup>-conforming N�d�lec element described in the PhD
+* thesis of S. Zaglmayr, <b>High Order Finite Element Methods
+* for Electromagnetic Field Computation</b>, Johannes Kepler
+* Univerit�t Linz, 2006.
+*
+* This element overcomes the sign conflict issues present in
+* traditional N�d�lec elements which arise from the edge and face
+* parameterisations used in the basis functions. Therefore, this
+* element should provide consistent results for general quadrilateral
+* and hexahedral elements.
+*
+* The way this element addresses the sign conflict problem is to assign
+* local edges and faces a globally defined orientation for all cells. The
+* local edge orientation is always chosen such that the first vertex defining
+* the edge is always chosen to be that which has the highest global vertex
+* numbering, with the second being that which has the lowest global vertex
+* numbering.
+*
+* Similarly, the face orientation is always chosen such that the first
+* vertex is chosen to be that with the highest global vertex numbering of the
+* four vertices making up the face. The third vertex is then chosen to be that
+* which is geometrically opposite the first vertex, and the second and fourth
+* vertices are decided such that the second has a higher global vertex numbering
+* than the fourth.
+*
+* Further details on this element, including some benchmarking, can be
+* in the paper R. Kynch, P. Ledger: <b>Resolving the sign conflict
+* problem for hp�hexahedral N�d�lec elements with application to eddy
+* current problems</b>, Computers & Structures 181, 41-54, 2017 (see
+* https://doi.org/10.1016/j.compstruc.2016.05.021).
+*
```

Comment:
Please say that the element does not cupport non-conforming meshes at the moment.